contrib/epee/src/net_utils_base.cpp

```diff
@@ -21,6 +21,19 @@ namespace epee { namespace net_utils
 	bool ipv4_network_address::is_loopback() const { return net_utils::is_ip_loopback(ip()); }
 	bool ipv4_network_address::is_local() const { return net_utils::is_ip_local(ip()); }
 
+	bool ipv6_network_address::equal(const ipv6_network_address& other) const noexcept
+	{ return is_same_host(other) && port() == other.port(); }
+
+	bool ipv6_network_address::less(const ipv6_network_address& other) const noexcept
+	{ return is_same_host(other) ? port() < other.port() : ip() < other.ip(); }
```

Comment:
Those will consider equal IPs different AFAICT, given the non canonical nature of the std::string ip field.

Reply:
changed to using boost's address_v6 under the hood